### PR TITLE
Add makefile and update readme accordingly

### PR DESCRIPTION
**Pull request by simondereuver**
# Makefile and update readme accordingly

## Summary
- Added Makefile for project with commands that can be read below and updated the README.md file with the same contents as below for clear instructions.

## Available Commands

1. **`make start`**:
   - **Purpose**: Sets up the environment by creating a virtual environment (if it doesn't already exist), installs the dependencies, opens VS Code, and prints instructions for manually activating the virtual environment.
   - **Steps**:
     1. Creates a virtual environment called `venv` (if necessary).
     2. Installs the project dependencies from `requirements.txt`.
     3. Opens the project in VS Code.
     4. Provides instructions to manually activate the virtual environment. (There is no workaround this, so the enviroment has to be manually activated after running `make start`).

   - **Usage**:
    `make start`

2. **`make lint`**:
    - **Purpose**: Runs pylint on the source code to check for code style issues and errors. This requires the virtual environment to be activated first, usefull to catch errors before you push and create a merge request (so it passes the pipeline).
    Note: You must manually activate the virtual environment before running this command, as the subshell created by Makefile doesn't persist the environment activation.

    - **Usage**:
    Activate the virtual enviroment:
    On Windows:
    `venv\Scripts\activate`
    On Unix-like systems (Linux, macOS):
    `source nameofenv/bin/activate`
    Then run: `make lint`.

3. **`make clean`**:

    - Purpose: Removes the virtual environment directory, cleaning up the project.
    - Steps:
    Removes the virtual environment folder **venv**.

    - **`Usage`**:
    `make clean`

4. **`IMPORTANT NOTE`**:
    - You need to have make installed to use it, makefile has been created with make version 4.4.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,6 +2,49 @@
...
+1. **`make start`**:
+   - **Purpose**: Sets up the environment by creating a virtual environment (if it doesn't already exist), installs the dependencies, opens VS Code, and prints instructions for manually activating the virtual environment.
+   - **Steps**:
+     1. Creates a virtual environment called `venv` (if necessary).
+     2. Installs the project dependencies from `requirements.txt`.
+     3. Opens the project in VS Code.
+     4. Provides instructions to manually activate the virtual environment. (There is no workaround this, so the enviroment has to be manually activated after running `make start`).
+
+   - **Usage**:
+    `make start`
```
Comment (ParnaSaeidpour) [inline README.md:20]: It doesn't work for me. I've run `make clean` first and then `make start` but got this error:
Screenshot 2024-09-12 at 13 19 48

Having a linter is useful, but this doesn't work for me. What was the system you tested on?

Tested on:
Macbook M3
OS: Sonoma 14.6.1

Comment (simondereuver) [reply README.md:20]: Yes, forgot to add in the documentation that I havent tested it on unix based systems.

Comment (simondereuver) [reply README.md:20]: I have only tested on windows so far.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+.PHONY: venv
+venv:
+ifeq ($(OSFLAG),windows)
+	if not exist $(VENV)\Scripts\activate (
+		python -m venv $(VENV)
+	)
+else
+	if [ ! -f "$(VENV)/bin/activate" ]; then
+		python3 -m venv $(VENV)
+	fi
```
Comment (ParnaSaeidpour) [inline Makefile:35]:
```suggestion
	if [ ! -f "$(VENV)/bin/activate" ]; then python3 -m venv $(VENV); fi
```
Apparently a missing semicolon, and my system doesn't like it when it is in multiple lines.

Comment (simondereuver) [reply Makefile:35]: I think its better if you test it on your system and push a solution, its more or less impossible for me to fix it, as I only have access to Windows and Linux, altough doing it for Linux would require me to set everything up as I havent used Linux for a while.

Comment (ParnaSaeidpour) [reply Makefile:35]: Please accept my suggestion, as I have resolved the issue in my system.

Comment (simondereuver) [reply Makefile:35]: Great, but I cant seem to find it, did you push it to my branch or where can I find it?

Comment (simondereuver) [reply Makefile]: It also seems like I forgot to push the changes I made yesterday in an attempt to fix the issue.

Comment (ParnaSaeidpour) [reply Makefile]: @simondereuver It's now works with the new changes you made

**Review by ParnaSaeidpour — APPROVED** [review]

**Pull request merged**